study/src/test/java/reflection/Junit4TestRunner.java

```diff
@@ -1,13 +1,26 @@
 package reflection;
 
+import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+import java.util.Arrays;
+
 class Junit4TestRunner {
 
     @Test
     void run() throws Exception {
         Class<Junit4Test> clazz = Junit4Test.class;
 
         // TODO Junit4Test에서 @MyTest 애노테이션이 있는 메소드 실행
+        Arrays.stream(clazz.getMethods())
+                .filter(method -> method.isAnnotationPresent(MyTest.class))
+                .forEach(method -> {
+                    try {
+                        Assertions.assertThatNoException()
+                            .isThrownBy(() -> method.invoke(clazz.getDeclaredConstructor().newInstance()));
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                });
     }
```

Comment:
```suggestion
    @Test
    void run() throws Exception {
        Class<Junit4Test> clazz = Junit4Test.class;

        // TODO Junit4Test에서 @MyTest 애노테이션이 있는 메소드 실행
        Arrays.stream(clazz.getMethods())
                .filter(method -> method.isAnnotationPresent(MyTest.class))
                .forEach(method -> assertNoException(clazz, method));
    }

    void assertNoException(Class<Junit4Test> clazz, Method method) {
        try {
            Assertions.assertThatNoException()
                    .isThrownBy(() -> method.invoke(clazz.getDeclaredConstructor().newInstance()));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
```

객체지향 생활 체조 1원칙이 indentation 깊이에 관한 원칙입니다.
리플렉션을 하다보면 try catch 를 해야할 경우가 많아서 의도치않게 깊어지는 경우가 자주 생기는데,
그럴 때는 이렇게 메서드 분리를 해주시면 좋습니다.